index.js

```diff
@@ -158,34 +166,49 @@ function addContext(url, context) {
 }
 
 async function createVcExamples() {
-  // generate base keypair and signature suite
-  const keyPairEd25519VerificationKey2020 = await Ed25519VerificationKey2020.generate();
-  const keyPairEd25519Multikey = await Ed25519Multikey.from(keyPairEd25519VerificationKey2020);
+  // generate base keypair and signature suites
+  // ecdsa-sd-2023
+  const keyPairEcdsaMultikeyKeyPair = await EcdsaMultikey
+    .generate({curve: 'P-256'});
+  const keyPairEcdsaMultiKey = await EcdsaMultikey
+    .from({...keyPairEcdsaMultikeyKeyPair});
+  const {createSignCryptosuite} = ecdsaSd2023Cryptosuite;
+  const suiteEcdsaMultiKey = new DataIntegrityProof({
+    signer: keyPairEcdsaMultiKey.signer(),
+    cryptosuite: createSignCryptosuite()
+  });
+  // Ed25519Signature2020
+  const keyPairEd25519VerificationKey2020 = await Ed25519VerificationKey2020
+    .generate();
+  const keyPairEd25519Multikey = await Ed25519Multikey
+    .from(keyPairEd25519VerificationKey2020);
   const suiteEd25519Signature2020 = new Ed25519Signature2020({
     key: keyPairEd25519VerificationKey2020
   });
+  // eddsa-rdfc-2022
   const suiteEd25519Multikey = new DataIntegrityProof({
-    signer: keyPairEd25519Multikey.signer(), cryptosuite: eddsaRdfc2022CryptoSuite
+    signer: keyPairEd25519Multikey.signer(),
+    cryptosuite: eddsaRdfc2022CryptoSuite
   });
+  // vc-jwt
   const jwk = await jose.generateKeyPair('ES256');
 
   // add styles for examples
   addVcExampleStyles();
 
   // process every example that needs a vc-proof
-  const vcProofExamples = document.querySelectorAll(".vc");
+  const vcProofExamples = document.querySelectorAll('.vc');
   let vcProofExampleIndex = 0;
   for(const example of vcProofExamples) {
     vcProofExampleIndex++;
 
-    const verificationMethod = example.dataset?.vcVm
-      || 'did:key:' + keyPairEd25519VerificationKey2020.publicKey;
+    const verificationMethod = example.dataset?.vcVm ||
+      'did:key:' + keyPairEd25519VerificationKey2020.publicKey;
 
```

Comment:
This isn't right because it always uses the Ed25519 key ... and it looks like it's adding not `publicKeyMultibase` but `publicKey`. Each verification method ID depends on the key used (different one for the Ed25519 key vs. the ECDSA key, etc.). Another option would be to make the verification method more opaque (like some HTTPS example URL) -- but then the VC becomes unverifiable as an example.

Note that if `did:key` is used, then the issuer ID should match the DID which should be `did:key:<publicKeyMultibase>` (importantly, not the public key ID, which should be `did:key:<publicKeyMultibase>#<publicKeyMultibase>`).

Another DID method or an HTTPS URL should be used to support a proof set. So if this is just generating one proof at a time, it's fine -- provided that the DIDs and VM IDs match up in each instance (for each cryptosuite).

Reply:
There were more bugs here than just misusing the same key. The should all be addressed now, though. Thanks!